server/users.ts

```diff
@@ -540,11 +542,21 @@ export class User extends Chat.MessageContext {
 		const auth = (room && !this.can('makeroom') ? room.auth.get(this.id) : this.group);
 		return auth in Config.groups && Config.groups[auth].rank >= Config.groups[minAuth].rank;
 	}
-	can(permission: RoomPermission, target: User | null, room: BasicRoom): boolean;
-	can(permission: GlobalPermission, target?: User | null): boolean;
-	can(permission: RoomPermission & GlobalPermission, target: User | null, room?: BasicRoom | null): boolean;
-	can(permission: string, target: User | null = null, room: BasicRoom | null = null): boolean {
-		return Auth.hasPermission(this, permission, target, room);
+	can(permission: RoomPermission, target: User | null, room: BasicRoom, useVisualGroup?: boolean): boolean;
+	can(permission: GlobalPermission, target?: User | null, room?: null, useVisualGroup?: boolean): boolean;
+	can(
+		permission: RoomPermission & GlobalPermission,
+		target: User | null,
+		room?: BasicRoom | null,
+		useVisualGroup?: boolean
+	): boolean;
+	can(
+		permission: string,
+		target: User | null = null,
+		room: BasicRoom | null = null,
+		useVisualGroup?: boolean
+	): boolean {
```

Comment:
Instead of overriding `user.can`, I think `this.can` should just call `Auth.hasPermission` directly.